### DISCUSSION TITLE
I can't write the proper code for mutation to update the table

### DISCUSSION THREAD

**Discussion by saifrahmn** [MichalLytek/type-graphql#1445] [Q&A] [answered]
I am getting an error while write the following code
```
@Mutation(() => Post, {nullable: true})
  async updatePost(
    @Arg("id") id: number,
    @Arg("title",() => String,{nullable: true}) title: string,
    @Ctx() { em }: MyContext
  ): Promise<Post[] | null> {
      const post= await em.find(Post, { id });
      if(!post){
          return null;
      }
      if (typeof title !== 'undefined'){
        post.title = title;
        await em.persistAndFlush(post);
      }
      return post;
  }
```

The error here occurs say the following
`Property 'title' does not exist on type 'Loaded<Post, never>[]'.`

I just want to update the title of a post, using the post id but it shows the above error

**Comment by carlocorradini** [comment]
This is an "error" of the persistence library (DB) and not a direct type-graphql one.
If you can add more information we can help you but without any knowledge on what library, etc... is hard ;/

**Reply by carlocorradini** [reply]
PS: Are you using Mikro ORM?
I've never tried it but take a look at the docs (maybe ref)?

**Reply by saifrahmn** [reply]
I am using mikro-orm, postgresql, typeqraphql

**Comment by saifrahmn** [comment, marked as answer]
Someone from reddit helped me, saying 

> em.find returns an array and not a single post. You should try something like findOne / findByPk

I fixed it my changing my const post variable
`const post= await em.findOne(Post, { id });`
So the answer is solved
